### PR TITLE
Feat : Add codesandbox storybook addon

### LINKED ISSUE #4563: Use CodeSandbox Storybook Add-on
CodeSandbox released a Storybook Add-on that allows users to select a story with specific props and open it directly in CodeSandbox with a live example that could be edited.

https://codesandbox.io/blog/announcing-the-codesandbox-storybook-add-on

https://github.com/recharts/recharts/assets/7189823/771b4e51-a911-43aa-af03-27a741952fa9

Currently, we create CodeSandbox links for individual stories that don't necessarily fit the described state. We should replace those hand-crafted CodeSandboxes with the Add-on mentioned above, which should provide better documentation and easier and better bug reports.

### PR DESCRIPTION
**Pull request by H0onnn**
## Description

Added the storybook addon provided by codesandbox.

## Related Issue

- close #4563

Not all issues have been resolved yet.

This is an issue related to environment variables, and more details can be found in the comments on the issue.

You need to add environment variables related to apiToken in the `storybook/preview.ts` file.

## Motivation and Context

By automating the creation of codesandbox links, users can more easily test the UI and bug fix.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: storybook/preview.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ const preview: Preview = {
     },
     // Sets a delay of 1550ms before each story to give them time to animate (recharts animation default is 1500ms)
     chromatic: { delay: 1550 },
+    codesandbox: {
+      apiToken: '',
```
Comment (PavelVanecek) [inline storybook/preview.ts:26]: So I would like to use this addon but I can't see how to get this token through. It's private so we cannot just embed it in the source code. Thoughts?

**Comment by PavelVanecek** [comment]
Also https://github.com/codesandbox/storybook-addon/issues/15 + https://github.com/codesandbox/storybook-addon/issues/17

File: storybook/preview.ts (hunk 1 above)
Comment (ckifer) [reply storybook/preview.ts:26]: I changed you to Admin in codesandbox - go to recharts -> settings -> permissions -> API Access

**Comment by ckifer** [comment]
https://github.com/codesandbox/storybook-addon/issues/17 might resolve once we finish refactoring but idk

**Comment by ckifer** [comment]
Going to close this, we can look into it again later

**Pull request closed without merging**